model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/conn/CustomChangeLogHistoryService.java

```diff
@@ -47,18 +50,29 @@ public class CustomChangeLogHistoryService extends StandardChangeLogHistoryServi
 
     private List<RanChangeSet> ranChangeSetList;
 
+    @Override
+    public boolean supports(Database database) {
+        return database instanceof MySQLDatabase;
+    }
+
+    @Override
+    public void init() throws DatabaseException {
+        super.init();
+
+        AddPrimaryKeyStatement pkStatement = new AddPrimaryKeyStatement(getLiquibaseCatalogName(), getLiquibaseSchemaName(), getDatabaseChangeLogTableName(), 
+                ColumnConfig.arrayFromNames("ID, AUTHOR, FILENAME"), "PK_DATABASECHANGELOG");
+        ChangelogJdbcMdcListener.execute(getDatabase(), ex -> ex.execute(pkStatement));
+        getDatabase().commit();
```

Comment:
I'm wondering if we should check if the database changelog table exists before adding the primary key, like we do in the `getRanChangeSets()` method bellow (the `if (hasDatabaseChangelogTable()) `.

Reply:
I believe it might be unnecessary, please correct me if I am wrong. 

It seems that after calling `StandardChangeLogHistoryService`'s init() method (`super.init();`) the table either already exists or is created within it or the exception is thrown. 

The `StandardChangeLogHistoryService` is guarded by `serviceInitialized` boolean flag to ensure the second call of it does nothing and I thing that is something what is missing in the PR. I'll update it.